Netable/Example/Requests/GraphQL/UpdatePostsMutation.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  UpdatePostsMutation.swift
+//  Example
+//
+//  Created by Amy Oulton on 2022-12-08.
+//  Copyright © 2022 Steamclock Software. All rights reserved.
+//
+
+import Foundation
+import Netable
+
+struct UpdatePostMutationInput: Codable {
+    let title: String
+    let content: String
+}
+
+struct GetAllPostsResponse: Decodable {
+    let posts: [Post]
+}
+
+struct UpdatePostsMutation: GraphQLRequest {
+    typealias Input = UpdatePostMutationInput
+    typealias RawResource = GetAllPostsResponse
+    typealias FinalResource = [Post]
+
+    var input: UpdatePostMutationInput
+    var source = GraphQLQuerySource.autoResource
+    var unredactedParameterKeys: Set<String> {
+        ["query", "variables"]
+    }
+
+    func finalize(raw: GetAllPostsResponse) async throws -> [Post] {
+        raw.posts
+    }
```

Comment:
Are we doing a manual finalize here to demonstrate how they work, or because smart unwrap doesn’t support this exact construction? If the former we should maybe add a comment saying that’s the case, if the latter, what is it that prevents it from working here?

Reply:
The former! I'll go ahead and add a comment above finalize specifying that.